main.go

```diff
@@ -43,26 +44,37 @@ func init() {
 	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
 
 	utilruntime.Must(lokiv1beta1.AddToScheme(scheme))
-	// +kubebuilder:scaffold:scheme
-
-	// TODO: make this loading optional via a loki-operator CLI flag.
-	utilruntime.Must(monitoringv1.AddToScheme(scheme))
 }
 
 func main() {
 	var metricsAddr string
 	var enableLeaderElection bool
 	var probeAddr string
+	var enableOpenshiftFeatures bool
+	var useTLSServiceMonitorConfig bool
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8080", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.BoolVar(&enableLeaderElection, "leader-elect", false,
 		"Enable leader election for controller manager. "+
 			"Enabling this will ensure there is only one active controller manager.")
+	flag.BoolVar(&enableOpenshiftFeatures, "with-ocp-features", false,
+		"Enables features in an Openshift cluster.")
+	flag.BoolVar(&useTLSServiceMonitorConfig, "with-tls-service-monitor-config", false,
+		"Enables loading of a prometheus service monitor.")
 	flag.Parse()
 
 	log.Init("loki-operator")
 	ctrl.SetLogger(log.GetLogger())
 
+	if useTLSServiceMonitorConfig {
+		// +kubebuilder:scaffold:scheme
```

Comment:
This annotation should stay in the `init` function to keep operator-sdk working.